app/javascript/packages/memorable-date/index.ts

```diff
@@ -0,0 +1,386 @@
+  min?: string;
+  max?: string;
+  message: string;
+}
+
+/**
+ * Type for a hash in which the specified messages can be looked up
+ */
+type MemorableDateErrorMessageLookup = Record<
+  MemorableDateErrorMessage & string,
+  string | undefined
+>;
+
+interface ErrorMessageLookupContainer {
+  error_messages: MemorableDateErrorMessageLookup;
+  range_errors: RangeErrorMessage[];
+}
+
+/**
+ * The MemorableDate custom HTML element (WebComponent) provides
+ * a broadly intuitive way for users to enter dates into web applications.
+ *
+ * More about the component here: https://designsystem.digital.gov/components/memorable-date/
+ *
+ * This class facilitates custom error checking and messaging for the MemorableDate
+ * (<lg-memorable-date />) in combination with the ValidatedFieldElement
+ * (<lg-validated-field />). The web server or another source is responsible for
+ * adding the expected child elements for use with this WebComponent.
+ */
+class MemorableDateElement extends HTMLElement {
+  /**
+   * HTML input element for entering the month part of the date
+   */
+  get monthInput(): HTMLInputElement | null {
+    return this.querySelector('.memorable-date__month');
+  }
+
+  /**
+   * HTML input element for entering the day part of the date
+   */
+  get dayInput(): HTMLInputElement | null {
+    return this.querySelector('.memorable-date__day');
+  }
+
+  /**
+   * HTML input element for entering the year part of the date
+   */
+  get yearInput(): HTMLInputElement | null {
+    return this.querySelector('.memorable-date__year');
+  }
+
+  /**
+   * List of HTML input elements for entering month, day and year for a date
+   */
+  get allInputs(): HTMLInputElement[] {
+    const month = this.monthInput;
+    const day = this.dayInput;
+    const year = this.yearInput;
+    const inputs: HTMLInputElement[] = [];
+
+    month && inputs.push(month);
+    day && inputs.push(day);
+    year && inputs.push(year);
+
+    return inputs;
+  }
+
+  /**
+   * The configured minimum valid value for the date, based on "min" HTML attribute
+   */
+  get min(): Date | null {
+    return this.getDateAttribute('min');
+  }
+
+  /**
+   * The configured maximum valid value for the date, based on "max" HTML attribute
+   */
+  get max(): Date | null {
+    return this.getDateAttribute('max');
+  }
+
+  connectedCallback() {
+    // Wait to show validation until first submission
+    let updateErrorMessage = false;
+
+    const { allInputs } = this;
+
+    const inputListener = (e: Event) => {
+      // Run validations
+      this.validate();
+
+      // Update error messages on form validation
+      if (e.type === 'invalid') {
+        updateErrorMessage = true;
+      }
+
+      if (updateErrorMessage) {
+        // Trigger error message updates
+        const errorMessage = allInputs.find((f) => f.validationMessage)?.validationMessage;
+        allInputs.forEach((f) => {
+          f.closest('lg-validated-field')?.setErrorMessage(errorMessage || null);
```

Comment:
I wouldn't expect memorable date to have any awareness of a validated field. We should be able to accomplish this with a combination of `#setCustomValidity` (to set the message) and optionally [#reportCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reportValidity) to notify the user, though I'm also unsure that we should be notifying the user as frequently about validation changes, since it departs from our standard approach of validating upon submission and may have accessibility implications (e.g. if notifications start to become distracting as the user fills out the field).

Reply:
I tried doing this in a decoupled way first, and that didn't turn out well.

The standard behavior on this page is to dismiss the error for a field when the user makes any change (even if it fails the exact same validation check). I'll remove this change so that the date field works the same way as the other fields.

Reply:
I updated this so that `memorable-date` doesn't interact directly with `lg-validated-field`. Since this left the borders on valid parts of the date, I used artificial `input` events to remove all error colors on change so that it acts more like the other input fields.

Reply:
This looks great now, thanks for those changes!